src/services/events/index.ts

```diff
@@ -0,0 +1,49 @@
+import { HeseyaEvents, EventCallbackFunction, GetTypeFromInterface } from './utils'
+
+type EventCallbackWrapper<T extends keyof HeseyaEvents> = EventCallbackFunction<
```

Comment:
Coś nie do końca z tymi typami tu jest, bo pozwala wyemitować event bez produktu 
![image](https://user-images.githubusercontent.com/26066123/182315428-1caf5674-2bb2-46cf-9765-706189370fe5.png)

Reply:
Nad tym trzeba jeszcze powalczyć, bo to w zasadzie najistotniejsze w tym całym mechanizmie

Może spróbuj się podpytać na kanale `Typescript` na mattermost, może ktoś coś podpowie